src/error.rs

```diff
@@ -52,3 +54,16 @@ impl From<i32> for Error {
         }
     }
 }
+
+impl <'a> std::fmt::Display for Error <'_> {
+    //TODO: Add documentation 
+    pub fn fmt(&self, f: &mut std::fmt::Formatter <'_> ) -> std::fmt::Result {
+        match *self {
+            Error::NoError(str: &str) => wirteln!(io::stderr(), "{}", str),
+            Error::OptionParsingError(str: &str) => wirteln!(io::stderr(), "{}", str),
```

Comment:
Errors should should be using `write!` instead of `writeln!`

```suggestion 
            Error::OptionParsingError(err: &str) => write!(io::stderr(), "{}", err),
```